### DISCUSSION TITLE
Cors issue in vite react app

### DISCUSSION THREAD

**Discussion by adityakumarcodes** [supabase/supabase#26369] [Questions]
![image](https://github.com/supabase/supabase/assets/71957886/2eeff364-7de8-4d6d-b1db-513749fc5d80)

Below component i am using ,but how to resolve the cors issue.
`import { useState, useEffect } from "react";
import Card from "../../atoms/Card";
import { supabase } from "../../../constants/supabaseClient";

const WordOfTheDay = () => {

    const [word, setWord] = useState("");
    const [counter, setCounter] = useState(0);

    useEffect(() => {
        getInfo();
    }, [counter]);

    const getInfo = async () => {
        try {
            const { data, error } = await supabase.from("words").select("*");

            console.log(data);
            if (error) throw error;
            if (data != null) {
                setWord(data[counter % data.length]);
            }
        } catch (error) {
            console.log(error);
        }
    };

    return <Card
        onClick={() => setCounter(counter + 1)}
        style={{
            backgroundImage:
                'linear-gradient(rgba(0, 0, 0, 0.527),rgba(0, 0, 0, 0.5)) ,url("https://images.unsplash.com/photo-1536147116438-62679a5e01f2?auto=format&fit=crop&q=60&w=600&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MTh8fGdyZWVufGVufDB8fDB8fHww") '
        }}
    >
        <p style={{ fontFamily: "Dancing Script", fontSize: "50px" }}>
            {word["word"]}
        </p>
        <div>
            <p className="text-xl text-balance">{word["meaning"]}</p>
        </div>
        <div style={{ marginTop: "50px" }}></div>
    </Card>;
}

export default WordOfTheDay;`

**Comment by lucasweaver10** [comment]
Hey there, did you ever get this fixed? If so mind sharing what your issue was?

**Reply by adityakumarcodes** [reply]
Nope ,I was using supabase as backend when facing the issue, now I moved to express backend to solve this issue

**Reply by rudrasaiprasad1** [reply]
> Hey there, did you ever get this fixed? If so mind sharing what your issue was?

I'm using React with Vite and Tailwind CSS for the frontend, and Supabase for the backend.

I encountered an issue that was resolved by correcting the prefix of the environment variables in Supabase. If you're using Vite as your build tool, ensure that all environment variables intended for the frontend are prefixed with VITE_. You can verify and update these variables in the Environment Variables section of your Supabase project settings.

This adjustment allowed my application to access the necessary configuration values correctly during build and runtime.
